src/Nano_transcriber.cpp

```diff
@@ -0,0 +1,168 @@
+                        ros::Publisher& long_range_pub,
+                        ros::Publisher& battery_publisher){
+
+    sensor_msgs::Range long_range_msg;
+    sensor_msgs::Range short_range_msg;
+    iarc7_msgs::FlowVector flow_msg;
+    iarc7_msgs::Float64Stamped battery_voltage;
+
+    long_range_msg.range = nano_info.long_range;
+    long_range_msg.min_range = 0.5;
+    long_range_msg.max_range = 10;
+    ros::Duration long_range_offset = ros::Duration().fromNSec((int64_t)(nano_info.long_range_offset * 10000));
+    long_range_msg.header.stamp = nano_info.msg_received - long_range_offset;
+    long_range_msg.radiation_type = sensor_msgs::Range::INFRARED;
+    long_range_msg.field_of_view = 0.3;
+    long_range_msg.header.frame_id = "/tfmini";
+    if(nano_info.long_range_offset > 0) {
+        long_range_pub.publish(long_range_msg);
+    }
+
+    short_range_msg.range = nano_info.short_range;
+    short_range_msg.min_range = 0.01;
+    short_range_msg.max_range = 0.800;
+    ros::Duration short_range_offset = ros::Duration().fromNSec((int64_t)(nano_info.short_range_offset * 10000));
+    short_range_msg.header.stamp =  nano_info.msg_received - short_range_offset;
+    short_range_msg.radiation_type = sensor_msgs::Range::INFRARED;
+    short_range_msg.field_of_view = 0.3;
+    short_range_msg.header.frame_id = "/short_distance_lidar";
+    if(nano_info.short_range_offset > 0)
+    {
+        rangefinder_pub.publish(short_range_msg);
+    }
+    
+    ros::Duration flow_offset = ros::Duration().fromNSec((int64_t)(nano_info.short_range_offset * 10000));
+    flow_msg.header.stamp = (nano_info.msg_received - flow_offset);
+    flow_msg.deltaX = nano_info.deltaX;
+    flow_msg.deltaY = nano_info.deltaY;
+    if(nano_info.flow_board_offset > 0)
+    {
+        opticalflow_pub.publish(flow_msg);
+    }
+
+    battery_voltage.data = nano_info.battery_voltage;
+    battery_voltage.header.stamp = nano_info.msg_received;
+    battery_publisher.publish(battery_voltage);
+
+    return;
+}
+
+void sendESCCommand(iarc7_msgs::PlanarThrottle esc_cmnds,
+                    ros::Publisher& esc_cmd_pub,
+                    bool activateSideRotors)
+{
+
+    iarc7_msgs::ESCCommand esc_cmd;
+
+    if(activateSideRotors)
+    {
+        esc_cmd.front_motor_PWM = (esc_cmnds.front_throttle*125) + 125;
+        esc_cmd.back_motor_PWM = (esc_cmnds.back_throttle*125) + 125;
+        esc_cmd.left_motor_PWM = (esc_cmnds.left_throttle*125) + 125;
+        esc_cmd.right_motor_PWM = (esc_cmnds.right_throttle*125) + 125;
+    }
+    else {
+        esc_cmd.front_motor_PWM = 125;
+        esc_cmd.back_motor_PWM = 125;
+        esc_cmd.left_motor_PWM = 125;
+        esc_cmd.right_motor_PWM = 125;
+    }
+
+    esc_cmd_pub.publish(esc_cmd);
+
+    return;
+
+}
+
+int main(int argc, char* argv[]){
+    
+    ros::init(argc, argv, "Nano_transcriber");
+    ros::NodeHandle nh;
+
+    ROS_WARN("Nano transcriber beginning");
```

Comment:
Make this a `ROS_INFO` or `ROS_DEBUG`

Reply:
Put that there when I couldn't get this to run at all and have not had that problem since I fixed it, do you care if I just delete it